compileopts/target.go

```diff
@@ -43,6 +43,7 @@ type TargetSpec struct {
 	FlashMethod      string   `json:"flash-method"`
 	FlashVolume      string   `json:"msd-volume-name"`
 	FlashFilename    string   `json:"msd-firmware-name"`
+	UF2FamilyID      uint32   `json:"uf2-family-id"`
```

Comment:
I think this property needs to be added to the copyProperties() method below, something like:
```
  if spec2.UF2FamilyID != 0 {                                                   
    spec.UF2FamilyID = spec2.UF2FamilyID                                                                                                                                  
  }
```
Otherwise the value of this property is lost in resolveInherits()

Reply:
Done

Reply:
Done. Now I'm wondering how this ever worked for me...

Reply:
I would suggest making this also a `string` and then using `hex.DecodeString()` or whatever to convert to the "real" type.

This just seems like it would make it a lot more readable in the actual JSON files themselves.

What do you think?

Reply:
Sounds good! and this allows us to distinguish between no family id (empty string) and a family id of zero!